conans/test/integration/toolchains/ios/test_using_cmake.py

```diff
@@ -0,0 +1,80 @@
+import platform
+import textwrap
+import unittest
+
+from conans.client.toolchain.cmake.base import CMakeToolchainBase
+from conans.test.utils.tools import TestClient
+from ._utils import create_library
+
+
+@unittest.skipUnless(platform.system() == "Darwin", "Requires XCode")
+class ToolchainiOSTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.t = TestClient()
+        create_library(self.t)
+        self._conanfile = textwrap.dedent("""
+            from conans import ConanFile, CMake, CMakeToolchain
+
+
+            class Library(ConanFile):
+                name = 'hello'
+                version = '1.0'
+                settings = 'os', 'arch', 'compiler', 'build_type'
+                exports_sources = 'hello.h', 'hello.cpp', 'cpp-wrapper.h', 'cpp-wrapper.mm', 'CMakeLists.txt'
+                options = {{'shared': [True, False]}}
+                default_options = {{'shared': False}}
+                _cmake = None
+
+                def _configure_cmake(self):
+                    if not self._cmake:
+                        self._cmake = CMake(self, generator={generator}, parallel=False)
+                        self._cmake.configure()
+                    return self._cmake
+
+                def toolchain(self):
+                    tc = CMakeToolchain(self)
+                    tc.write_toolchain_files()
+
+                def build(self):
+                    cmake = self._configure_cmake()
+                    cmake.configure()
+                    cmake.build()
+                    self.run("lipo -info Release-iphoneos/libhello.a")
+
+                def package(self):
+                    cmake = self._configure_cmake()
+                    cmake.install()
+            """)
+
+        self.t.save({
+            'ios_profile': textwrap.dedent("""
+                [settings]
+                os=iOS
+                os.version=12.0
+                arch=armv8
+                compiler=apple-clang
+                compiler.version=12.0
+                compiler.libcxx=libc++
+                build_type=Release
+            """)
+        })
+
+    def test_xcode_generator(self):
+        """ Simplest approach:
+            https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-ios-tvos-or-watchos
+        """
+        self.t.save({'conanfile.py': self._conanfile.format(generator='"Xcode"')})
```

Comment:
To discuss what would be a way so the recipe doesn't need to hardcode ``generator=Xcode`` in it, so it can be generic to more platforms.